src/vse_sync_pp/analyzers/pmc.py

```diff
@@ -0,0 +1,129 @@
+    STATE_HOLDOVER_OUT_OF_SPEC2: handle_holdover_out_of_spec_transition,
+    STATE_HOLDOVER_OUT_OF_SPEC3: handle_holdover_out_of_spec_transition,
+}
+
+FREERUN_TRANSITION = {
+    STATE_FREERUN: STATE_FREERUN,
+    STATE_LOCKED: STATE_LOCKED,
+}
+
+LOCKED_TRANSITION = {
+    STATE_FREERUN: STATE_FREERUN,
+    STATE_LOCKED: STATE_LOCKED,
+    STATE_HOLDOVER_IN_SPEC: STATE_HOLDOVER_IN_SPEC,
+}
+
+HOLDOVER_IN_SPEC_TRANSITION = {
+    STATE_LOCKED: STATE_LOCKED,
+    STATE_HOLDOVER_IN_SPEC: STATE_HOLDOVER_IN_SPEC,
+    STATE_HOLDOVER_OUT_OF_SPEC1: STATE_HOLDOVER_OUT_OF_SPEC1,
+    STATE_HOLDOVER_OUT_OF_SPEC2: STATE_HOLDOVER_OUT_OF_SPEC2,
+    STATE_HOLDOVER_OUT_OF_SPEC3: STATE_HOLDOVER_OUT_OF_SPEC3,
+}
+
+HOLDOVER_OUT_OF_SPEC_TRANSITION = {
+    STATE_HOLDOVER_IN_SPEC: STATE_HOLDOVER_IN_SPEC,
+    STATE_HOLDOVER_OUT_OF_SPEC1: STATE_HOLDOVER_OUT_OF_SPEC1,
+    STATE_HOLDOVER_OUT_OF_SPEC2: STATE_HOLDOVER_OUT_OF_SPEC2,
+    STATE_HOLDOVER_OUT_OF_SPEC3: STATE_HOLDOVER_OUT_OF_SPEC3,
+}
+
+class ClockStateAnalyzer(Analyzer):
+    """Analyze clock state
+    """
+    id_ = 'phc/gm-settings'
+    parser = id_
+    def __init__(self, config):
+        super().__init__(config)
+        # samples in the initial transient period are ignored
+        self._transient = config.parameter('transient-period/s')
+        # minimum test duration for a valid test
+        self._duration = config.parameter('min-test-duration/s')
+        self.state = None
+    def prepare(self, rows):
+        idx = 0
+        try:
+            tstart = rows[0].timestamp + self._transient
+        except IndexError:
+            pass
+        else:
+            while idx < len(rows):
+                if tstart <= rows[idx].timestamp:
+                    break
+                idx += 1
+        return super().prepare(rows[idx:])
+    def test(self, data):
+        if len(data) == 0:
+            return (False, "no data")
+        
+        for index, row in data.iterrows():
+            clock_class = row['clock_class']
+ 
+            if (self.state is None) and (clock_class in STATE_TRANSITION):
+                self.state = clock_class
+            else:
+                try:
+                    transitioned = STATE_TRANSITION[self.state](self, clock_class)
+                except KeyError:
+                    return (False, f"wrong clock class {clock_class}")
+                
+                if transitioned is not None:
+                    return (False, "wrong state transition")
+
+        if data.iloc[-1].timestamp - data.iloc[0].timestamp < self._duration:
+            return (False, "short test duration")
+        if len(data) - 1 < self._duration:
+            return (False, "short test samples")
+        return (True, None)
+    def explain(self, data):
+        if len(data) == 0:
+            return {}
+        return {
+            'duration': data.iloc[-1].timestamp - data.iloc[0].timestamp,
+            'state': self.state,
```

Comment:
Instead of anouncing the last state I propose  to add a list of pairs (duration, state) where we report all the states PMC has announced during `duration`.

Reply:
I agree it should be something other than the last state.
What value do we get with a list of pairs? Trying to think about the worst case of a flapping state, which will give us a big list.

Reply:
What about reporting the number of transitions over the full duration? That seems relevant to the intent of the analyser (are all transitions right), gives some more insight into the dataset without being too noisy if state is flapping. For a bit more detail output the number of times each state was observed in the duration (and possibly total duration spent in that state.)

Reply:
So @crwr45 you read my mind :-)
What I would do:
1) return the total number of samples for each of the states (freerun, locked, holdover_in_spec, holdover_out_of_spec)
2) return the total number of transitions during the whole duration

This wont give us all the information about state transitions and duration that we need but I agree that is not the place to put that information. Because the flapping Nati mentioned could perfectly happen and still be compliant with the T-GM state transition spec. 

In any case we don't want to lose that information reporting the evolution of state during time. I think the best way to represent evolution over time of the different state transition will have to be reported by the plot module of vse-sync-pp (x axis time, y axis state).

Last thing, I'm gonna refine the test spec associated to teh pmc-analyzer to add the following the attributes that are already gathered by the pmc collector:  ClockAccuracy, and OffsetScaledLogVariance.

Reply:
I added the number of states and the total number. Currently, the output looks like this
```json
{"result": true, "reason": null, "analysis": {"duration": 1999.0688657, "clock_class_count": {"FREERUN": 0, "LOCKED": 501, "HOLDOVER_IN_SPEC": 1008, "HOLDOVER_OUT_SPEC": 488}, "total_transitions": 2}}
```

Maybe we can rename it a bit if you think it's needed.
@jnunyez What would you like to see for the other information like the `ClockAccuracy`? just the last one?

Reply:
Nothing to see as result when things go well, we will use them to check for inconsistencies.
We will use `ClockAccuracy` and `OffsetScaledLogVariance`  in collected values. to ensure the T-GM PTP messages are announcing the proper `ClockAccuracy` value and `OffsetScaledLogVariance` values associated to the `ClockClass`.  (Take a look at step 4 in TTINTEL-285.) For instance if collector reports `ClockState` is LOCKED but `ClockAccuracy` is 0xFE (meaning my clock is not connected to a PRTC) we should report an error.

Reply:
I think it should be added to the PMC parser first. Not sure if it should be part of this PR

Reply:
Fair enough.